SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityS3UploaderDriver.java

```diff
@@ -187,6 +188,17 @@ public void startAndWait() {
       }
     }, configuration.getCheckUploadsEverySeconds(), configuration.getCheckUploadsEverySeconds(), TimeUnit.SECONDS);
 
+    fileSyncFuture = scheduler.scheduleAtFixedRate(() -> {
+      runLock.lock();
+      try {
+        readInitialFiles();
```

Comment:
checking - do you know how long this process usually takes, and is there a concern that it'll hold the lock long enough to cause issues for the S3 uploads?

Reply:
This whole process is async/background, nothing hits it live or like a service, and it only holds the lock for a second or two